ideascube/serveradmin/catalog.py

```diff
@@ -831,14 +832,18 @@ def _progress(*args):
                     self._progress(' {}'.format(remote.name), *args)
 
                 # TODO: Verify the download with sha256sum? Crypto signature?
-                try:
-                    urlretrieve(remote.url, tmppath, reporthook=_progress)
-                except ConnectionError:
-                    print("Warning: Impossible to connect to the remote"
-                          " {remote.name}({remote.url}).\n"
-                          "Continue anyway without this remote."
-                          .format(remote=remote))
-                    continue
+                urlparsed = urlparse(remote.url)
+                if urlparsed.scheme in ['file', '']:
+                    shutil.copyfile(urlparsed.path, tmppath)
```

Comment:
copyfile may raise exceptions.
You should try/except them to handle them correctly (as we do when we use urlretrieve).